subsys/bluetooth/host/classic/avdtp.c

```diff
@@ -30,50 +30,69 @@ LOG_MODULE_REGISTER(bt_avdtp);
 #define AVDTP_MSG_POISTION 0x00
 #define AVDTP_PKT_POSITION 0x02
 #define AVDTP_TID_POSITION 0x04
-#define AVDTP_SIGID_MASK 0x3f
+#define AVDTP_SIGID_MASK   0x3f
 
-#define AVDTP_GET_TR_ID(hdr) ((hdr & 0xf0) >> AVDTP_TID_POSITION)
+#define AVDTP_GET_TR_ID(hdr)    ((hdr & 0xf0) >> AVDTP_TID_POSITION)
 #define AVDTP_GET_MSG_TYPE(hdr) (hdr & 0x03)
 #define AVDTP_GET_PKT_TYPE(hdr) ((hdr & 0x0c) >> AVDTP_PKT_POSITION)
-#define AVDTP_GET_SIG_ID(s) (s & AVDTP_SIGID_MASK)
+#define AVDTP_GET_SIG_ID(s)     (s & AVDTP_SIGID_MASK)
 
 static struct bt_avdtp_event_cb *event_cb;
 static sys_slist_t seps;
 
 #define AVDTP_CHAN(_ch) CONTAINER_OF(_ch, struct bt_avdtp, br_chan.chan)
 
-#define AVDTP_KWORK(_work) CONTAINER_OF(CONTAINER_OF(_work, struct k_work_delayable, work),\
-					struct bt_avdtp, timeout_work)
+#define AVDTP_KWORK(_work)                                                                         \
+	CONTAINER_OF(CONTAINER_OF(_work, struct k_work_delayable, work), struct bt_avdtp,          \
+		     timeout_work)
 
 #define DISCOVER_REQ(_req) CONTAINER_OF(_req, struct bt_avdtp_discover_params, req)
-#define GET_CAP_REQ(_req) CONTAINER_OF(_req, struct bt_avdtp_get_capabilities_params, req)
+#define GET_CAP_REQ(_req)  CONTAINER_OF(_req, struct bt_avdtp_get_capabilities_params, req)
 #define SET_CONF_REQ(_req) CONTAINER_OF(_req, struct bt_avdtp_set_configuration_params, req)
-#define OPEN_REQ(_req) CONTAINER_OF(_req, struct bt_avdtp_open_params, req)
-#define START_REQ(_req) CONTAINER_OF(_req, struct bt_avdtp_start_params, req)
+#define CTRL_REQ(_req)     CONTAINER_OF(_req, struct bt_avdtp_ctrl_params, req)
 
 #define AVDTP_TIMEOUT K_SECONDS(6)
 
-K_MUTEX_DEFINE(avdtp_mutex);
-#define AVDTP_LOCK() k_mutex_lock(&avdtp_mutex, K_FOREVER)
-#define AVDTP_UNLOCK() k_mutex_unlock(&avdtp_mutex)
+#define AVDTP_LOCK()   k_sem_take(&session->sem_lock, K_FOREVER)
+#define AVDTP_UNLOCK() k_sem_give(&session->sem_lock)
```

Comment:
Please never create macros that make this kind of assumptions of the environment that they're invoked in, i.e. in this case that there has to be a given variable with a specific name. You should instead pass the session as a macro parameter. That said, I'm not sure what's the benefit of using macros here, since I don't think you want to preserve the possibility of redefining them or similar. Can't these just be one-line functions (which the compiler will certainly inline)?
```c
static void avdtp_lock(struct bt_avdtp *session)
{
	k_sem_take(&session->sem_lock);
}
```

Reply:
Updated as you suggested.